webpack.config.js

```diff
@@ -50,16 +48,40 @@ function requestToExternal(request) {
   return undefined;
 }
 
+const isProduction = process.env.NODE_ENV === 'production';
+const mode = isProduction ? 'production' : 'development';
+
 const sharedConfig = {
+  mode,
+  devtool: !isProduction ? 'source-map' : undefined,
   output: {
     path: path.resolve(process.cwd(), 'assets', 'js'),
     filename: '[name].js',
     chunkFilename: '[name].js',
   },
   module: {
-    ...defaultConfig.module,
     rules: [
-      ...defaultConfig.module.rules,
+      !isProduction && {
+        test: /\.js$/,
+        use: ['source-map-loader'],
+        enforce: 'pre',
+      },
```

Comment:
You could spread this to avoid the filter below:

```js
      ...(!isProduction && [{
        test: /\.js$/,
        use: ['source-map-loader'],
        enforce: 'pre',
      }]),
```

As `...false` does nothing.

Though I don't really know which is nicer.